systemtest/src/main/java/io/strimzi/systemtest/performance/utils/TopicOperatorPerformanceUtils.java

```diff
@@ -0,0 +1,299 @@
+        KafkaTopicScalabilityUtils.createTopicsViaK8s(testStorage.getNamespaceName(), testStorage.getClusterName(),
+            testStorage.getTopicName(), start, end, 12, 3, 2);
+        KafkaTopicScalabilityUtils.waitForTopicStatus(testStorage.getNamespaceName(), testStorage.getTopicName(),
+            start, end, CustomResourceStatus.Ready, ConditionStatus.True);
+    }
+
+    private static void performModificationWithWait(int start, int end, ExtensionContext currentContext, TestStorage testStorage, Map<String, Object> kafkaTopicConfigToModify) {
+        ResourceManager.setTestContext(currentContext);
+        LOGGER.info("Modifying Kafka topics from index {} to {}", start, end);
+        KafkaTopicScalabilityUtils.modifyBigAmountOfTopics(testStorage.getNamespaceName(), testStorage.getTopicName(),
+            start, end, new KafkaTopicSpecBuilder().withConfig(kafkaTopicConfigToModify).build());
+        KafkaTopicScalabilityUtils.waitForTopicsContainConfig(testStorage.getNamespaceName(), testStorage.getTopicName(),
+            start, end, kafkaTopicConfigToModify);
+    }
+
+    private static void performDeletionWithWait(int start, int end, ExtensionContext currentContext, TestStorage testStorage) {
+        ResourceManager.setTestContext(currentContext);
+        LOGGER.info("Deleting Kafka topics from index {} to {}", start, end);
+        deleteKafkaTopicsInRange(testStorage.getNamespaceName(), testStorage.getTopicName(), start, end);
+        waitForTopicWithPrefixDeletion(testStorage.getNamespaceName(), testStorage.getTopicName(), start, end);
+    }
+
+    /**
+     * Deletes Kafka topics within a specified range based on their names.
+     * Assumes topic names are structured to include a numerical suffix indicating their batch.
+     *
+     * @param namespace     the Kubernetes namespace
+     * @param prefix        the common prefix of the Kafka topic names
+     * @param start         the starting index of topics to delete
+     * @param end           the ending index of topics to delete
+     */
+    public static void deleteKafkaTopicsInRange(String namespace, String prefix, int start, int end) {
+        List<KafkaTopic> topicsToDelete = KafkaTopicResource.kafkaTopicClient().inNamespace(namespace).list().getItems()
+            .stream()
+            .filter(topic -> {
+                String name = topic.getMetadata().getName();
+                if (!name.startsWith(prefix)) return false;
+                // Extract index from the topic name
+                String indexPart = name.substring(prefix.length());
+                try {
+                    int index = Integer.parseInt(indexPart);
+                    return index >= start && index < end;
+                } catch (NumberFormatException e) {
+                    LOGGER.error("Failed to parse index from Kafka topic name: {}", name, e);
+                    return false;
+                }
+            }).toList();
```

Comment:
I guess this is something that would be useful also somewhere else? At least I would not bind it just to `performance`

Reply:
Cool, I would move it to KafkaTopicUtils